### PR TITLE
Reset timer instead of `start()` when stepDown

### LINKED ISSUE #407: Election timer should be restarted instead of being started.
### Describe the bug

https://github.com/sofastack/sofa-jraft/blob/master/jraft-core/src/main/java/com/alipay/sofa/jraft/core/NodeImpl.java#L1269

We should restart the `electionTimer` instead of starting it, the reasons for it are:
 1. This can make the code clearer. According to the codebase here, it's hard to make sure what the timer behavior is when it triggers a timeout since we concat 2 intervals to one.

2. Image that `self` is a follower,  and the election timer is not triggered now(assume there are m milliseconds to timeout.). Now we receive a current valid leader from others, `stepDown()` is invoked, and `electionTimer.start()` is invoked as well, We DO NOT reset the timer. So the next election timeout is `ELECTION_TIMEOUT - m`, which does not make sense because the election timeout should be a reasonable range which we defined before.

### How to fix
The fix of this issue could be very simple, just use `timer.restart()` instead of `timer.start()`.

### PR DESCRIPTION
**Pull request by jovany-wang**
This PR closes #407 .

### CODE REVIEW

**Comment by jovany-wang** [comment]
I have signed CLA, but why it still failed?

File: jraft-core/src/main/java/com/alipay/sofa/jraft/util/RepeatedTimer.java [hunk 1]
```diff
@@ -151,6 +151,33 @@ public void start() {
...
+                this.running = true;
+                schedule();
+                return;
+            }
+            if (this.running) {
+                // Reschedule it.
+                schedule();
+                return;
+            }
+        } finally {
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/util/RepeatedTimer.java]: Is this code
```
    if (this.stopped) {
                this.stopped = false;
                this.running = true;
                schedule();
                return;
            }
            if (this.running) {
                // Reschedule it.
                schedule();
                return;
            }
```

can simplify to 

```
this.stopped = false;
this.running = true;
schedule();
```

？

File: jraft-core/src/main/java/com/alipay/sofa/jraft/util/RepeatedTimer.java [hunk 2]
```diff
@@ -151,6 +151,33 @@ public void start() {
         }
     }
 
+    /**
+     * Restart the timer.
+     *
+     * It will be start if it's stopped, and it will be restart if it's running.
+     */
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/util/RepeatedTimer.java]: add @author?

**Comment by jovany-wang** [comment]
@fengjiachun All comments were addressed.

**Pull request merged**